atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/KeyValueService.java

```diff
@@ -373,6 +373,17 @@ void putUnlessExists(@QueryParam("tableRef") TableReference tableRef,
     @Idempotent
     void deleteRange(@QueryParam("tableRef") TableReference tableRef, RangeRequest range);
 
+    /**
+     * For each cell, deletes all timestamps prior to the associated maximum timestamp, excluding garbage collection
+     * sentinels. Depending on the implementation, this may result in a range tombstone in the underlying KVS.
+     */
+    @POST
+    @Path("delete-all-timestamps")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Idempotent
+    void deleteAllTimestamps(@QueryParam("tableRef") TableReference tableRef,
+            Map<Cell, Long> maxTimestampExclusiveByCell);
```

Comment:
Is it worth adding a serialisation test for this? (do we ever actually expect someone outside Atlas/Nimbus to hit this?)

Reply:
A serialization test for this endpoint? I don't think anyone uses these endpoints